### PR TITLE
Add slider customizations

### PR DESCRIPTION
**Pull request by maryx**
When creating a Slider widget, a user can now customize:
- inactiveColor, the color of the not-yet-played portion of the slider (default to theme.unselectedWidgetColor)
- showThumb, whether to show the circle thumb (default to true)

### CODE REVIEW

**Comment by Hixie** [comment]
This will need tests; the `test/rendering/mock_canvas.dart` file has a `paints` matcher which may help.

**Comment by Hixie** [comment]
What's the use case for not showing the thumb?

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -118,6 +118,95 @@ void main() {
...
+    await tester.pumpWidget(buildApp(null));
+
+    final RenderBox sliderBox =
+        tester.firstRenderObject<RenderBox>(find.byType(Slider));
+
+    expect(sliderBox, paints..rect(color: theme.accentColor)..rect(color: theme.unselectedWidgetColor));
+    expect(sliderBox, paints..circle(color: theme.accentColor));
+    await tester.pumpWidget(buildApp(customColor));
+    expect(sliderBox, paints..rect(color: customColor)..rect(color: theme.unselectedWidgetColor));
+    expect(sliderBox, paints..circle(color: customColor));
```
Comment (Hixie) [inline packages/flutter/test/material/slider_test.dart]: you probably want some negative tests here too, verifying that there isn't a circle of the custom color when the custom color isn't set and that there isn't one of the accent color when the custom color _is_ set.

**Comment by maryx** [comment]
One of our use cases of the slider doesn't have the thumb. However, after some discussion, we are removing that use case, so I will remove the `showThumb` from this PR.

File: packages/flutter/test/material/slider_test.dart (hunk 1 above)
Comment (maryx) [inline packages/flutter/test/material/slider_test.dart]: Added.

**Comment by Hixie** [comment]
Ok!

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
This can land once the tree is green.

**Comment by Hixie** [comment]
Thank you very much for your contribution!

**Comment by sethladd** [comment]
Thank you @maryx !

**Comment by maryx** [comment]
@Hixie Thanks very much! Can you merge this in please? I don't have write access.

**Comment by Hixie** [comment]
Done! Thanks again!

**Pull request merged**